fedimint-bitcoind/src/lib.rs

```diff
@@ -198,22 +198,28 @@ impl DynBitcoindRpc {
     pub fn spawn_block_count_update_task(
         self,
         task_group: &TaskGroup,
-    ) -> anyhow::Result<watch::Receiver<Option<u64>>> {
-        let (block_count_tx, block_count_rx) = watch::channel(None);
+        on_update: impl Fn(u64) + Send + Sync + 'static,
+    ) -> anyhow::Result<()> {
         let mut desired_interval = get_bitcoin_polling_interval();
 
+        // Note: atomic only to workaround Send+Sync async closure limitation
+        let last_block_count = AtomicU64::new(0);
+
         task_group.spawn_cancellable("block count background task", {
             async move {
-                debug!(target: LOG_BITCOIND, "Updating bitcoin block count");
+                trace!(target: LOG_BITCOIND, "Fetching block count from bitcoind");
 
                 let update_block_count = || async {
                     let res = self
                         .get_block_count()
                         .await;
 
                     match res {
-                        Ok(c) => {
-                            let _ = block_count_tx.send(Some(c));
+                        Ok(block_count) => {
+                            if last_block_count.load(Ordering::SeqCst) != block_count {
```

Comment:
[nit] Is there a reason we can't use `Ordering::Relaxed`? (Same question for rest of PR)

Reply:
`SeqCst` is safest, most conservative option, so I don't even want to think about it. :)